### ISSUE TITLE
Add geo: URI when sharing

### ISSUE THREAD

**Issue by StyXman** [organicmaps/organicmaps#1367]
Currently if you share a location, OM includes a `om://` URI and a link to omaps.app, but no `geo:` URI. This means that whoever that receives the location can't chose what application to use to handle the shared location.

**Comment by Misalf-git** [comment]
You can still simply copy only the GPS coordinates for pasting in any app.
However, I very much dislike the long text provided by the share location feature as it pretends to be my own wording while it's more likely it was intended to convince others to use the app.
```
Hey, sieh dir meinen aktuellen Standort auf der Organic Maps-Karte an - Hi,

ich bin gerade hier: Mein Standort. Klicke den Link om://xxxxxxxxxx/Mein_Standort oder https://omaps.app/xxxxxxxxxx/Mein_Standort, um den Ort auf der Karte anzuzeigen.

Vielen Dank.
```

**Comment by StyXman** [comment]
Doing that on a phone is not so easy. Clicking on a `geo:` link is super easy :)

**Comment by Misalf-git** [comment]
Ah, OK. I misunderstood.

**Comment by biodranik** [comment]
geo: coordinates are not automatically highlighted by apps. If you know any which app highlights the text like `geo:12.1234,33.12312` please let us know. It is possible to make a clickable link only when sharing and HTML text with embedded `<a href="geo:12.213,23.3123">Link</a>` links in it.

**Comment by kuba-orlik** [comment]
Sometimes the intention is nit to send the URL to a location in a text message or email, but to open the selected coordinates in ankther navigation app. For that, an optiob to generate a geo: URL would be very helpful.

**Comment by biodranik** [comment]
To open geo url in another app you need either a clickable html link, or copy coordinates and paste them into another app. A plain text geo: link won't be clickable.

**Comment by kuba-orlik** [comment]
A geo: link can be opened with another app without clicking - we can initiate an intent to open the URL and Android will list all the apps that support that. So maybe instead of changing the text when using the share button we could have another button: "open in another app"? 🤔

**Comment by StyXman** [comment]
> geo: coordinates are not automatically highlighted by apps

I know, I've been opening issues in many projects and tracking them here:

https://wiki.openstreetmap.org/wiki/Geo_URI_scheme

**Comment by biodranik** [comment]
> A geo: link can be opened with another app without clicking - we can initiate an intent to open the URL and Android will list all the apps that support that. So maybe instead of changing the text when using the share button we could have another button: "open in another app"? 🤔

That may be useful in some cases when you don't need to send a link to someone else. But how to implement a good UX/UI for it without complications?

**Comment by biodranik** [comment]
@StyXman great idea with a Wiki page! We've started but not finished it here: #1821

**Comment by kuba-orlik** [comment]
@biodranik

> That may be useful in some cases when you don't need to send a link to someone else. But how to implement a good UX/UI for it without complications?

We could have a "share" button and "open in external app" button

**Comment by biodranik** [comment]
Please think _carefully_ about the UX (User Experience) before any suggestions. Now there are 4 or 5 buttons on the bottom toolbar, where the last one is "share". Do you propose adding _one more_ button there, so all users will see 5 or 6 buttons at the bottom toolbar, out of which two (share and open in external app) will be rarely used?

**Comment by StyXman** [comment]
> Do you propose adding one more button [?]

Not really, no, I think adding the geo: URI to the text of the 'share position' function.

**Comment by StyXman** [comment]
I think the user story was not clear.

Sharing the position requires four steps:

* The map app (in this case OM) creates a text to send and picks up and application to do it
* That application sends the message to the other party.
* The same app (in case of closed gardens) or another one receives the message and recognizes the geo: URI.
* When the use clicks on it, their own map app (OM or another) opens the map in the right place.

What I'm asking is about that first step.

**Comment by contrapunctus-1** [comment]
I too would find it useful to share geo: URIs. Use case is opening a location for editing in Vespucci. 

It's possible to copy-paste the coordinates into Vespucci, but that's quite a few more steps than the share action workflow. I use OsmAnd for this, but that's not a workflow I can suggest to others, since OsmAnd is unusably slow on low end devices.

The geo: link share action could be placed somewhere in the "swipe up" menu, e.g. in the menu that comes up when long-pressing the coordinates.

Alternately, do as Osmand does - the share button (in OM's bottom bar) can open a menu in which you select what to share (e.g. "Share with a person" for the current links and text, and "Open in app" for the geo: link).

**Comment by StyXman** [comment]
> @StyXman great idea with a Wiki page! We've started but not finished it here: #1821

Great, except they switched to some Slavic+cirylic language and I can't understand a thing :(
